test/checkpointer_tests.jl

```diff
@@ -0,0 +1,39 @@
+using ClimaCore: Meshes, Domains, Topologies, Spaces, Fields, InputOutput
+using ClimaCoupler: TestHelper
+using ClimaComms
+using Test
+import ClimaCoupler: Interfacer
+import ClimaCoupler.Checkpointer: get_model_state_vector, set_checkpoint_model_state!, checkpoint_model_state
+
+FT = Float64
+
+struct DummySimulation{S} <: Interfacer.AtmosModelSimulation
+    state::S
+end
+get_model_state_vector(sim::DummySimulation) = sim.state
+
+@testset "get_model_state_vector" begin
+    boundary_space = TestHelper.create_space(FT)
+    sim = DummySimulation((; T = ones(boundary_space)))
+    @test get_model_state_vector(sim) == sim.state
+
+    sim2 = Interfacer.SurfaceStub([])
+    @test get_model_state_vector(sim2) == nothing
+end
+
+@testset "checkpoint_model_state, set_checkpoint_model_state!" begin
+    boundary_space = TestHelper.create_space(FT)
+    t = 1.0
+    comms_ctx = ClimaComms.context(ClimaComms.CPUSingleThreaded())
+    # old sim run
+    sim = DummySimulation(Fields.FieldVector(T = ones(boundary_space)))
+    checkpoint_model_state(sim, comms_ctx, t, output_dir = "test_checkpoint")
+
+    # new sim run
+    sim_new = DummySimulation(Fields.FieldVector(T = zeros(boundary_space)))
+    set_checkpoint_model_state!(sim_new, t, output_dir = "test_checkpoint")
+    @test sim_new.state.T == sim.state.T
+
+    # remove checkpoint directory
+    rm("./test_checkpoint/", force = true, recursive = true)
```

Comment:
Tests seem reasonable; there are entries for all 3 added functions. Minor comment -  in staying with the `checkpoint-restart` terminology would it be useful to perhaps `set_checkpoint_model_state -> restart_model_state` ?

Reply:
Sure, good idea!